src/WebJobs.Script/Workers/Rpc/Configuration/WorkerConfigurationResolverOptionsSetup.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.Rpc.Configuration
+{
+    internal sealed class WorkerConfigurationResolverOptionsSetup : IConfigureOptions<WorkerConfigurationResolverOptions>
+    {
+        private readonly IConfiguration _configuration;
+        private readonly IScriptHostManager _scriptHostManager;
+
+        public WorkerConfigurationResolverOptionsSetup(IConfiguration configuration, IScriptHostManager scriptHostManager)
+        {
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+            _scriptHostManager = scriptHostManager ?? throw new ArgumentNullException(nameof(scriptHostManager));
+        }
+
+        public void Configure(WorkerConfigurationResolverOptions options)
+        {
+            var configuration = _configuration;
+            if (_scriptHostManager is IServiceProvider scriptHostManagerServiceProvider)
+            {
+                var latestConfiguration = scriptHostManagerServiceProvider.GetService<IConfiguration>();
+                if (latestConfiguration is not null)
+                {
+                    configuration = new ConfigurationBuilder()
+                        .AddConfiguration(_configuration)
+                        .AddConfiguration(latestConfiguration)
+                        .Build();
+                }
```

Comment:
Configuration is setup similar to `LanguageWorkerOptionsSetup` - https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Workers/Rpc/Configuration/LanguageWorkerOptionsSetup.cs#L63-L71

Reply:
Nothing consumes this configuration. Why is it being built? In `LanguageWorkerOptionsSetup` it is being passed to `RpcWorkerConfigFactory`, so it is actually used.

Reply:
line 38 in this class consumes this configuration -

`options.WorkersDirPath = GetWorkersDirPath(configuration);`